src/sentry/incidents/models/incident.py

```diff
@@ -0,0 +1,395 @@
+        or the current time if it's still open.
+        """
+        return self.date_closed if self.date_closed else timezone.now()
+
+    @property
+    def duration(self):
+        return self.current_end_date - self.date_started
+
+    def normalize_before_relocation_import(
+        self, pk_map: PrimaryKeyMap, scope: ImportScope, flags: ImportFlags
+    ) -> int | None:
+        old_pk = super().normalize_before_relocation_import(pk_map, scope, flags)
+        if old_pk is None:
+            return None
+
+        # Generate a new UUID, if one exists.
+        if self.detection_uuid:
+            self.detection_uuid = uuid4()
+        return old_pk
+
+
+@region_silo_only_model
+class PendingIncidentSnapshot(Model):
+    __relocation_scope__ = RelocationScope.Organization
+
+    incident = OneToOneCascadeDeletes("sentry.Incident", db_constraint=False)
+    target_run_date = models.DateTimeField(db_index=True, default=timezone.now)
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_pendingincidentsnapshot"
+
+
+@region_silo_only_model
+class IncidentSnapshot(Model):
+    __relocation_scope__ = RelocationScope.Organization
+
+    incident = OneToOneCascadeDeletes("sentry.Incident", db_constraint=False)
+    event_stats_snapshot = FlexibleForeignKey("sentry.TimeSeriesSnapshot", db_constraint=False)
+    unique_users = models.IntegerField()
+    total_events = models.IntegerField()
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_incidentsnapshot"
+
+
+@region_silo_only_model
+class TimeSeriesSnapshot(Model):
+    __relocation_scope__ = RelocationScope.Organization
+    __relocation_dependencies__ = {"sentry.Incident"}
+
+    start = models.DateTimeField()
+    end = models.DateTimeField()
+    values = ArrayField(of=ArrayField(models.FloatField()))
+    period = models.IntegerField()
+    date_added = models.DateTimeField(default=timezone.now)
+
+    class Meta:
+        app_label = "sentry"
+        db_table = "sentry_timeseriessnapshot"
+
+    @classmethod
+    def query_for_relocation_export(cls, q: models.Q, pk_map: PrimaryKeyMap) -> models.Q:
+        pks = IncidentSnapshot.objects.filter(
+            incident__in=pk_map.get_pks(get_model_name(Incident))
+        ).values_list("event_stats_snapshot_id", flat=True)
+
+        return q & models.Q(pk__in=pks)
+
+
+class IncidentActivityType(Enum):
+    CREATED = 1
+    STATUS_CHANGE = 2
+    COMMENT = 3
+    DETECTED = 4
+
+
+@region_silo_only_model
+class IncidentActivity(Model):
+    __relocation_scope__ = RelocationScope.Organization
+
+    incident = FlexibleForeignKey("sentry.Incident")
+    user_id = HybridCloudForeignKey(settings.AUTH_USER_MODEL, on_delete="CASCADE", null=True)
+    type: models.Field[int | IncidentActivityType, int] = models.IntegerField()
```

Comment:
🤔 

Why would this line be perfectly fine in the singular `models` file - however now that it's split out into a separate file, mypy i snow angry at the type???

I'm seeing

```
error: Incompatible types in assignment (expression has type "IntegerField[float | int | str | Combinable, int]", variable has type "Field[int | IncidentActivityType, int]")  [assignment]
```

and i'm not entirely sure how to get around it...
importing `Combinable` seems to be having issues as well 🤔

Reply:
Not sure if related, but you're missing an `__init__.py` from your models folder.

Reply:
yeah, i think i've seen this before with missing __init__'s